### PR TITLE
Fix while project opening OnGeneratedCSProject has thrown null exception

### LINKED ISSUE #40: [BUG] Open project will get null exception
### Describe the bug

Open project will get null exception.

### To Reproduce

1. Create unity project
2. Add NuGet-importer-for-Unity package via OpenUPM
3. Unity console shown as follows:

```
ArgumentNullException: Value cannot be null.
Parameter name: path1
System.IO.Path.Combine (System.String path1, System.String path2) (at <c2a97e0383e8404c9fc0ae19d58f57f1>:0)
kumaS.NuGetImporter.Editor.NuGetImporterSettings.get_Instance () (at Library/PackageCache/org.kumas.nuget-importer@2.3.1/Editor/NuGetImporterSettings.cs:36)
kumaS.NuGetImporter.Editor.NuGetAnalyzerImportSetting.OnGeneratedCSProject (System.String path, System.String content) (at Library/PackageCache/org.kumas.nuget-importer@2.3.1/Editor/NuGetAnalyzerImportSetting.cs:111)
System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <c2a97e0383e8404c9fc0ae19d58f57f1>:0)
Rethrow as TargetInvocationException: Exception has been thrown by the target of an invocation.
System.Reflection.RuntimeMethodInfo.Invoke (System.Object obj, System.Reflection.BindingFlags invokeAttr, System.Reflection.Binder binder, System.Object[] parameters, System.Globalization.CultureInfo culture) (at <c2a97e0383e8404c9fc0ae19d58f57f1>:0)
System.Reflection.MethodBase.Invoke (System.Object obj, System.Object[] parameters) (at <c2a97e0383e8404c9fc0ae19d58f57f1>:0)
Packages.Rider.Editor.ProjectGeneration.ProjectGeneration.OnGeneratedCSProject (System.String path, System.String content, System.Type[] types) (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/ProjectGeneration/ProjectGeneration.cs:369)
Packages.Rider.Editor.ProjectGeneration.ProjectGeneration.SyncProjectFileIfNotChanged (System.String path, System.String newContents, System.Type[] types) (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/ProjectGeneration/ProjectGeneration.cs:296)
Packages.Rider.Editor.ProjectGeneration.ProjectGeneration.SyncProject (Packages.Rider.Editor.ProjectGeneration.ProjectPart island, System.Type[] types) (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/ProjectGeneration/ProjectGeneration.cs:286)
Packages.Rider.Editor.ProjectGeneration.ProjectGeneration.GenerateAndWriteSolutionAndProjects (System.Type[] types) (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/ProjectGeneration/ProjectGeneration.cs:220)
Packages.Rider.Editor.ProjectGeneration.ProjectGeneration.Sync () (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/ProjectGeneration/ProjectGeneration.cs:144)
Packages.Rider.Editor.RiderScriptEditor.Initialize (System.String editorInstallationPath) (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/RiderScriptEditor.cs:262)
Unity.CodeEditor.CodeEditor.SetCodeEditor (System.String editorPath) (at <5f40cdb07bd44d76a23dad985a4ec283>:0)
Unity.CodeEditor.CodeEditor.Register (Unity.CodeEditor.IExternalCodeEditor externalCodeEditor) (at <5f40cdb07bd44d76a23dad985a4ec283>:0)
Packages.Rider.Editor.RiderScriptEditor..cctor () (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/RiderScriptEditor.cs:29)
UnityEngine.Debug:LogException(Exception)
Packages.Rider.Editor.RiderScriptEditor:.cctor() (at Library/PackageCache/com.unity.ide.rider@3.0.15/Rider/Editor/RiderScriptEditor.cs:100)
UnityEditor.EditorAssemblies:ProcessInitializeOnLoadAttributes(Type[])
```

### Expected behavior

All clean, everything is happy.

### Desktop (please complete the following information):

 - OS: Windows 11 21H2
 - Unity Version 2021.3.8f1
 - Package Version 2.3.1

### PR DESCRIPTION
**Pull request by chika-chang**
Fixed #40

### CODE REVIEW

**Comment by kumaS-nu** [comment]
Thank you for the PR suggestion.
There is a problem with this change. (I should have written appropriate comments.)

## problem

The member `projectSettingsPath` is static, so it is initialized the first time the class `NuGetImporterSettings` is called. Also, since this package is multi-threaded, the class can be called from other threads than the main thread. However, Unity's `Application.dataPath` can only be called from the main thread. Therefore, an error will raise in environments where `OnGeneratedCSProject` is not called (for example, when using VS).

## What should we do

- Revert current changes
- Change `SetProjectSettingsPath` to `protected` so that it can be called explicitly.
- Call `SetProjectSettingsPath` before where `NuGetImporterSettings.Instance` may be called before setting the path (NuGetAnalyzerImportSetting.cs L.111, PackageManager.cs L.231).

File: NuGetImporterForUnity/Packages/NuGet Importer/Editor/NuGetImporterSettings.cs [hunk 1]
```diff
@@ -16,13 +16,7 @@ public class NuGetImporterSettings
         private static NuGetImporterSettings instance;
 
         [NonSerialized]
-        private static string projectSettingsPath;
-
-        [InitializeOnLoadMethod]
-        private static void SetProjectSettingsPath()
```
Comment (kumaS-nu) [inline NuGetImporterForUnity/Packages/NuGet Importer/Editor/NuGetImporterSettings.cs:22]: Change from private to protected

**Review by kumaS-nu — CHANGES_REQUESTED** [review]
- Revert current changes
- Call `SetProjectSettingsPath()`

**Comment by chika-chang** [comment]
protected can not be accessed from anther class without inheritance.

I add ensure method with internal access modifier.
Make sure `projectSettingsPath` is set before `NuGetImporterSettings.Instance` been called (NuGetAnalyzerImportSetting.cs L.111, PackageManager.cs L.231).

**Comment by kumaS-nu** [comment]
Thank you for your response. 
I made the elementary mistake of mistaking `internal` for `protected`...

This change looks good.
I will merge this PR after the test passes.

**Review by kumaS-nu — COMMENTED** [review]
This change looks good. Thank you!

**Pull request merged**